scripts/extract/check_new_date.py

```diff
@@ -52,13 +51,31 @@ def prepare_df_to_append(new_sitting_dates, version=2):
 ### Main Run ###
 
 
-def process(seeds_date_filepath):
-    df = pd.read_csv(seeds_date_filepath)
+def process():
+
+    # read from gbq instead of from filepath
+
+    df = pandas_gbq.read_gbq(query_or_table = """
+                                    SELECT *
+                                    FROM `singapore-parliament-speeches.raw.dates`
+                                    """)
 
     unchecked_dates_df = unchecked_dates(last_date_checked(df["Sitting_Date"]))
+
     new_sitting_dates_list = new_parliament_sitting_dates(unchecked_dates_df)
 
     append_df = prepare_df_to_append(new_sitting_dates_list, version)
-    append_df.to_csv(seeds_date_filepath, mode="a", index=False, header=False)
 
-    return 0
+    new_dates_df = pd.concat([df, append_df], ignore_index = True)
+
+    # check for duplicates
+    new_dates_df = new_dates_df.drop_duplicates(subset='Sitting_Date')
```

Comment:
what could be the risks of having a duplicate? (as the logic in the functions `unchecked_dates` and `last_date_checked` should not give rise to duplicates)

Reply:
this is what we talked about, an extra step to make sure there are no duplicate dates before uploading

Reply:
nvm i changed the logic